lib/cog/models/group.ex

```diff
@@ -4,17 +4,16 @@ defmodule Cog.Models.Group do
 
   alias Cog.Models.GroupRole
   alias Cog.Models.UserGroupMembership
+  alias Cog.Models.User
+  alias Cog.Models.Role
 
   alias Ecto.Changeset
 
   schema "groups" do
     field :name, :string
 
-    has_many :user_membership, UserGroupMembership, foreign_key: :group_id
-    has_many :direct_user_members, through: [:user_membership, :member]
-
-    has_many :role_grants, GroupRole
-    has_many :roles, through: [:role_grants, :role]
+    many_to_many :users, User, join_through: UserGroupMembership, join_keys: [group_id: :id, member_id: :id]
```

Comment:
ugh, so because of the way the groupable bits are implemented I think changing this to `user_id` would involve a bigger refactor. I'm gonna leave it for now and we can revisit if it ends up being too confusing for folks.